### PR TITLE
diff reduction for 24024

### PR DESCRIPTION
**Pull request by jbrockmendel**
docstrings, comments, edits that are correct both before and after #24024

the only substantive change is adding a test specifically for constructing a DatetimeArray from a PandasArray.

### CODE REVIEW

**Review by gfyoung — COMMENTED** [review]
Truth be told, as someone who hasn't looked too closely at #24024, it's hard to understand what you mean by "correct" regarding these changes.  By itself, it would have been tricky to understand the motivation.

That being said, as you are pulling this out of an existing PR, I'll trust that these will mesh with what you're ultimately aiming to do there.

**Review by jreback — CHANGES_REQUESTED** [review]
small comments. ping on green.

File: pandas/tests/frame/test_indexing.py [hunk 1]
```diff
@@ -3245,7 +3245,8 @@ def test_setitem(self):
         b1 = df._data.blocks[1]
         b2 = df._data.blocks[2]
         assert b1.values.equals(b2.values)
-        assert id(b1.values.values.base) != id(b2.values.values.base)
+        if b1.values.values.base is not None:
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/tests/frame/test_indexing.py]: should be covered by the comment 2 lines up

**Comment by TomAugspurger** [comment]
Again, lodging a request that we stop breaking pieces off.

**Comment by jbrockmendel** [comment]
> Again, lodging a request that we stop breaking pieces off.

Aside from a diff-reduction PR made on your branch moments before this request came in: done.

**Comment by jreback** [comment]
@TomAugspurger close this an proceed with #24024 ? ok by me

**Comment by jbrockmendel** [comment]
This is pretty benign and cuts 10% off of 24024

**Comment by jbrockmendel** [comment]
plus a pretty test

**Comment by TomAugspurger** [comment]
I haven't looked at this yet, so I can't choose in this specific case. Just
stating a general preference.

**Comment by jbrockmendel** [comment]
ping

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
ok, over to you @TomAugspurger for #24024

**Pull request merged**